src/gen.rs

```diff
@@ -225,7 +235,7 @@ fn gen_unmangle_method(ctx: &mut GenCtx,
                         span: ctx.span,
                         global: false,
                         segments: vec!(ast::PathSegment {
-                            identifier: ctx.ext_cx.ident_of(&v.mangled),
+                            identifier: ctx.ext_cx.ident_of(&mangled_rs),
```

Comment:
Not sure if the generated function already does `rust_id(xxx)`, it seems plausible (I'll test later). If not, it should also be used there.